File: security-management-frontend/src/services/ArticleContratTravailService.js
// src/services/ArticleContratTravailService.js
import api from "./api";

const RESOURCE = "/articles-contrat-travail";  // Corrigé pour éviter la duplication de /api/

export default {
    // Liste de tous les articles
    getAll: () => api.get(`${RESOURCE}`),
    
    // Détail d'un article par son ID
    getById: id => api.get(`${RESOURCE}/${id}`),
    
    // Tous les articles rattachés à un contrat de travail
    getByContratTravail: contratId => api.get(`${RESOURCE}/contrat-travail/${contratId}`),
    
    // Pour la compatibilité avec le code existant
    getByContratId: contratId => api.get(`${RESOURCE}/contrat-travail/${contratId}`),
    
    // Création
    create: article => api.post(RESOURCE, article),
    
    // Mise à jour
    update: (id, article) => api.put(`${RESOURCE}/${id}`, article),
    
    // Suppression
    remove: id => api.delete(`${RESOURCE}/${id}`)
};

File: SecurityManagementApp/src/main/java/com/boulevardsecurity/securitymanagementapp/service/ArticleContratTravailService.java
// src/main/java/com/boulevardsecurity/securitymanagementapp/service/ArticleContratTravailService.java
package com.boulevardsecurity.securitymanagementapp.service;

import com.boulevardsecurity.securitymanagementapp.dto.ArticleContratTravailCreationDto;
import com.boulevardsecurity.securitymanagementapp.dto.ArticleContratTravailDto;

import java.util.List;
import java.util.Optional;

public interface ArticleContratTravailService {

    List<ArticleContratTravailDto> getAll();

    Optional<ArticleContratTravailDto> getById(Long id);

    List<ArticleContratTravailDto> getByContratTravail(Long contratTravailId);

    ArticleContratTravailDto create(ArticleContratTravailCreationDto creationDto);

    ArticleContratTravailDto update(Long id, ArticleContratTravailCreationDto updateDto);

    void delete(Long id);
}
